src/style/design-tokens/carbon-scoped-tokens-provider/carbon-scoped-tokens-provider.component.js

```diff
@@ -21,32 +21,25 @@ const kebabCase = (input) =>
 export const tokensClassName = (theme) =>
   `sage-design-tokens-${kebabCase(theme)}-${carbonInstanceId}`;
 
-export const GlobalTokens = createGlobalStyle`
-  ${({ theme }) => {
-    const className = tokensClassName(theme.name);
-    activeThemes[className] =
-      activeThemes[className] || generateCssVariables(theme.compatibility);
-
-    return Object.entries(activeThemes).reduce(
-      (acc, [name, definitions]) => `${acc} .${name} { ${definitions} }`,
-      ""
-    );
-  }}`;
-
-const TokensProviderWrapper = styled.div.attrs(({ theme }) => ({
-  className: tokensClassName(theme.name),
-}))`
+export const TokensProviderWrapper = styled.div`
   margin: 0;
   padding: 0;
   width: auto;
   display: inline;
+
+  ${({ theme }) => generateCssVariables(theme.compatibility)}
 `;
 
+TokensProviderWrapper.defaultProps = {
+  theme: baseTheme,
+};
+
 const CarbonScopedTokensProvider = ({ children }) => (
```

Comment:
I think we could delete `TokensProviderWrapper` and add the styles to `CarbonScopedTokensProvider` to reduce the number of components.